lib/PDF/FPDF/TFPDFEngine.php

```diff
@@ -0,0 +1,139 @@
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+namespace SuiteCRM\PDF\FPDF;
+
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+
+use SuiteCRM\PDF\PDFEngine;
+use tFPDF;
+
+/**
+ * Class TFPDFEngine
+ * @package SuiteCRM\PDF\FPDF
+ */
+class TFPDFEngine extends PDFEngine
+{
+    /**
+     * @var tFPDF
+     */
+    public $pdf;
+
+    /**
+     * @var string
+     */
+    private static $configMapperFile = __DIR__ . '/../../../lib/PDF/MPDF/configMapping.php';
+
+    /**
+     * TFPDFEngine constructor.
+     * @param tFPDF|null $pdf
+     */
+    public function __construct(tFPDF $pdf = null)
+    {
+        $this->pdf = $pdf ?? new tFPDF();
+    }
+
+    /**
+     * @param string $html
+     * @param int $section 0=default; 1=headerCSS; 2=HTML body; 3=HTML parses; 4=HTML headers;
+     * @return void
+     */
+    public function writeHTML(string $html, int $section = 0): void
+    {
+        $this->pdf->Write($section, $html);
+    }
+
+    /**
+     * @param string $name
+     * @param string $destination
+     * @return void|string
+     */
+    public function outputPDF(string $name, string $destination): ?string
+    {
+        $output = $this->pdf->Output($name, $destination);
+
+        if (is_string($output)) {
+            return $output;
+        }
+    }
+
+    /**
+     * @param string|array $html
+     * @return void
+     */
+    public function writeHeader(string $html): void
+    {
+        $this->pdf->Write(0, $html);
+    }
+
+    /**
+     * @param string|array $html
+     * @return void
+     */
+    public function writeFooter(string $html): void
+    {
+        $this->pdf->Write(0, $html);
```

Comment:
Hi @Dillon-Brown `writeFooter` and `writeHeader` have the same implementation is that correct?

Reply:
Yup, instead of defining sections the header is just called first and footer called last.

Reply:
ok thank you